tests/D2L.CodeStyle.Analyzers.Test/Specs/ImmutabilityAnalyzer.cs

```diff
@@ -1223,3 +1223,125 @@ public SomeRecord( SomeRecord v, int w, Types.SomeImmutableClass x, int y, objec
+		: ISomethingImmutable { }
+
+	public struct
+		/* MissingTransitiveImmutableAttribute(ConsistencyTests.SadStructImplementor, , interface, ConsistencyTests.ISomethingImmutable) */ SadStructImplementor /**/
+		: ISomethingImmutable { }
+
+	public sealed class
+		/* MissingTransitiveImmutableAttribute(ConsistencyTests.SadDeriver, , base class, ConsistencyTests.HappyImplementor) */ SadDeriver /**/
+		: HappyImplementor { }
+
+	public sealed class
+		/* MissingTransitiveImmutableAttribute(ConsistencyTests.SadImplementor2, , interface, ConsistencyTests.ISomethingImmutable) */ SadImplementor2 /**/
+		: VanillaBase, IVanilla, IVanilla2, ISomethingImmutable { }
+
+	public sealed class
+		/* MissingTransitiveImmutableAttribute(ConsistencyTests.SadImplementor3, , base class, ConsistencyTests.HappyImplementor) */ SadImplementor3 /**/
+		: HappyImplementor, IVanilla, IVanilla2 { }
+
+	public interface
+		/* MissingTransitiveImmutableAttribute(ConsistencyTests.SadExtender, , interface, ConsistencyTests.ISomethingImmutable) */ SadExtender /**/
+		: ISomethingImmutable { }
+
+	// This won't emit an error because SadDeriver hasn't added [Immutable].
+	// There is a diagnostic for that mistake, but once its fixed we would
+	// get a diagnostic here. It would be nicer to developers to report all
+	// the violations, probably, but this keeps the implementation very simple
+	// and it's unlikely to come up in practice if you make small changes
+	// between compiles.
+	public sealed class IndirectlySadClass : SadDeriver { }
+
+	public partial class
+	/* MissingTransitiveImmutableAttribute(ConsistencyTests.PartialClass, , interface, ConsistencyTests.ISomethingImmutable) */ PartialClass /**/
+		: ISomethingImmutable { }
+
+	// This one doesn't get the diagnostic. We attach it to the one that specified
+	// the interface.
+	public partial class PartialClass { }
+
+	// We will emit another diagnostic here though... this makes sense but the
+	// code fix will try to apply multiple [Immutable] attributes which isn't
+	// allowed...
+	public partial class
+	/* MissingTransitiveImmutableAttribute(ConsistencyTests.PartialClass, , base class, ConsistencyTests.HappyImplementor) */ PartialClass /**/
+		: HappyImplementor { }
+
+	// This shouldn't crash the analyzer
+	public sealed class Foo : IThingThatDoesntExist { }
+
+	[Immutable]
+	public record UnsealedImmutableRecord { }
+
+	public sealed record
+		/* MissingTransitiveImmutableAttribute(ConsistencyTests.DerivedRecordMissingAttribute, , base class, ConsistencyTests.UnsealedImmutableRecord) */ DerivedRecordMissingAttribute /**/
+		: UnsealedImmutableRecord { }
+
+	[Immutable]
+	public sealed record SealedDerivedWithAttribute : UnsealedImmutableRecord { }
+
+	[Immutable]
+	public record UnsealedDerivedWithAttribute : UnsealedImmutableRecord { }
+
+	public record RegularRecord { }
+	public sealed record RegularDerivedRecord : RegularRecord { }

```

Comment:
Some adjustments had to be made below this. It previously read `public sealed RegularDerivedRecord : RegularRecord { }` (i.e. missing the `record` keyword), which caused the namespace to syntactically end.

This wasn't a problem in the source of the tests since the `using static Objects` was outside of the namespace.